code/modules/projectiles/projectile/bullets.dm

```diff
@@ -65,6 +65,28 @@
 
 	return 0
 
+/* foam dart projectiles */
+
+/obj/item/projectile/bullet/foam_dart
+	name = "foam dart"
+	desc = "I hope you're wearing eye protection."
+	damage = 0 // It's a damn toy.
+	damage_type = OXY
+	nodamage = TRUE
+	icon = 'icons/obj/gun_toy.dmi'
+	icon_state = "foamdart_proj"
+	range = 15
+	impact_effect_type = null
+	fire_sound = 'sound/items/syringeproj.ogg'
+
+/obj/item/projectile/bullet/foam_dart/on_impact(var/atom/A)
+	. = ..()
+	new /obj/item/ammo_casing/afoam_dart(get_turf(loc))

```

Comment:
Possibly worth putting in a `var/turf/T = get_turf(loc);;if(istype(T))` before this and on_range in case of hitting an atom that deletes the projectile (I believe the supermatter does this but not sure)